flax/core/frozen_dict.py

```diff
@@ -74,40 +66,54 @@ def copy(self, **add_or_replace):
     return type(self)(self, **add_or_replace)
 
   def items(self):
-    self._ensure_dict()
-    return self._dict.items()
+    for key in self._dict:
+      yield (key, self[key])
 
-  def _ensure_dict(self):
-    if self._dict is None:
-      self._dict = jax.tree_unflatten(self._treedef, self._leaves)
+  def unfreeze(self) -> Dict[K, V]:
+    return unfreeze(self)
 
   def tree_flatten(self):
-    if self._treedef is None:
-      self._leaves, self._treedef = jax.tree_flatten(self._dict)
-    return self._leaves, self._treedef
+    return (self._dict,), ()
 
   @classmethod
-  def tree_unflatten(cls, aux_data, children):
-    return cls(aux_data, children)
+  def tree_unflatten(cls, _, data):
+    return cls(*data)
 
 
-def freeze(x: Dict[K, V]) -> FrozenDict[K, V]:
-  """Freeze a nested dict."""
-  if not isinstance(x, dict):
-    return x
-  temp = {}
-  for key, value in x.items():
-    temp[key] = freeze(value)
-  return FrozenDict(temp)
+def _freeze(xs: Mapping[K, V]) -> FrozenDict[K, V]:
```

Comment:
Ah. I see now. So this method name doesn't sound right. It's actually `_unfreeze`?

Reply:
Thanks your comment made me realize `_freeze` is redundant